metricflow/dataflow/optimizer/source_scan/cm_branch_combiner.py

```diff
@@ -240,8 +240,8 @@ def visit_aggregate_measures_node(  # noqa: D
         combined_parent_node = combined_parent_nodes[0]
         assert combined_parent_node is not None
 
-        combined_metric_input_measure_specs = (
-            self._current_left_node.metric_input_measure_specs + current_right_node.metric_input_measure_specs
+        combined_metric_input_measure_specs = tuple(
```

Comment:
Since the iteration order of the set is not defined, can you use a dict instead?